app/controllers/social_previews_controller.rb

```diff
@@ -2,16 +2,22 @@ class SocialPreviewsController < ApplicationController
   # No authorization required for entirely public controller
 
   PNG_CSS = "body { transform: scale(0.3); } .preview-div-wrapper { overflow: unset; margin: 5vw; }".freeze
-  SHE_CODED_TAGS = %w[shecoded theycoded shecodedally].freeze
 
   def article
     @article = Article.find(params[:id])
     @tag_badges = Badge.where(id: Tag.where(name: @article.decorate.cached_tag_list_array).pluck(:badge_id))
     not_found unless @article.published
 
-    template = (@article.decorate.cached_tag_list_array & SHE_CODED_TAGS).any? ? "shecoded" : "article"
+    template = @article.tags.
+      where("tags.social_preview_template IS NOT NULL AND tags.social_preview_template != ?", "article").
```

Comment:
It's not clear how to handle a situation when an article is tagged with several tags with a custom template. For now, I choose the first one of these tags.

Reply:
@benhalpern thoughts here?